trino/sqlalchemy/dialect.py

```diff
@@ -0,0 +1,301 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from textwrap import dedent
+from typing import Any, Dict, List, Optional, Tuple
+
+from sqlalchemy import exc, sql
+from sqlalchemy.engine.base import Connection
+from sqlalchemy.engine.default import DefaultDialect
+from sqlalchemy.engine.url import URL
+
+from trino import dbapi as trino_dbapi
+from trino.auth import BasicAuthentication
+
+from . import compiler, datatype, error
+
+
+class TrinoDialect(DefaultDialect):
+    name = 'trino'
+    driver = 'rest'
+
+    statement_compiler = compiler.TrinoSQLCompiler
+    ddl_compiler = compiler.TrinoDDLCompiler
+    type_compiler = compiler.TrinoTypeCompiler
+    preparer = compiler.TrinoIdentifierPreparer
+
+    # Data Type
+    supports_native_enum = False
+    supports_native_boolean = True
+    supports_native_decimal = True
+
+    # Column options
+    supports_sequences = False
+    supports_comments = True
+    inline_comments = True
+    supports_default_values = False
+
+    # DDL
+    supports_alter = True
+
+    # DML
+    supports_empty_insert = False
+    supports_multivalues_insert = True
+    postfetch_lastrowid = False
+
+    @classmethod
+    def dbapi(cls):
+        """
+        ref: https://www.python.org/dev/peps/pep-0249/#module-interface
+        """
+        return trino_dbapi
+
+    def create_connect_args(self, url: URL) -> Tuple[List[Any], Dict[str, Any]]:
+        args, kwargs = super(TrinoDialect, self).create_connect_args(url)  # type: List[Any], Dict[str, Any]
+
+        db_parts = kwargs.pop('database', 'hive').split('/')
```

Comment:
Because [PyHive use this value](https://github.com/dropbox/PyHive/blob/b21c507a24ed2f2b0cf15b0b6abb1c43f31d3ee0/pyhive/sqlalchemy_presto.py#L93). But I just merge a PR to change database default value to `system`.

Reply:
Is a default database required? i.e. what would fail if no default database was set?
The dbapi implementation doesn't set any default database today.
Note that the user may not have permission to access whatever default catalog we pick.

If it's required then `system` seems like the correct choice because it'll always exist.

Reply:
I have change default database to `system`